packages/react/src/components/Popup/PopupContent.tsx

```diff
@@ -123,7 +142,15 @@ class PopupContent extends UIComponent<WithAsProp<PopupContentProps>> {
             },
           },
         )}
-      </ElementType>
+      </>
+    )
+
+    return trapFocus ? (
+      <FocusTrapZone {...focusTrapZoneProps}>{popupContent}</FocusTrapZone>
+    ) : autoFocus ? (
+      <AutoFocusZone {...autoFocusZoneProps}>{popupContent}</AutoFocusZone>
+    ) : (
+      <ElementType {...popupContentProps}>{popupContent}</ElementType>
```

Comment:
I suggest to refactor it in a bit more obvious way:

```jsx
   const popupContentProps: PopupContentProps = {
      className: classes.root,
      ...rtlTextContainer.getAttributes({ forElements: [children, content] }),
      ...accessibility.attributes.root,
      ...unhandledProps,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
    }

    const popupContent = (
      <>
        {pointing && (
          <Ref innerRef={pointerRef}>
            {Box.create({}, { defaultProps: { styles: styles.pointer } })}
          </Ref>
        )}
        {Box.create(
          {},
          {
            defaultProps: {
              children: childrenExist(children) ? children : content,
              styles: styles.content,
            },
          },
        )}
      </>
    )
    
    if (trapFocus) {
      const focusTrapZoneProps = {
        ...popupContentProps,
        ...((_.keys(trapFocus).length && trapFocus) as FocusTrapZoneProps),
        as: ElementType,
      }
      
      return <FocusTrapZone {...focusTrapZoneProps}>{popupContent}</FocusTrapZone> 
    }

    if (autoFocus) {
      const autoFocusZoneProps = {
        ...popupContentProps,
        ...((_.keys(autoFocus).length && autoFocus) as AutoFocusZoneProps),
        as: ElementType,
      }
      
      return <AutoFocusZone {...autoFocusZoneProps}>{popupContent}</AutoFocusZone>
    }
    
    return <ElementType {...popupContentProps}>{popupContent}</ElementType>
```

Reply:
1. We will compute `focusTrapZoneProps` and `autoFocusZoneProps` only when they are really needed
2. We will avoid nested ternaries

Reply:
❤️